main/src/library/Map.flix

```diff
@@ -115,17 +117,10 @@ namespace Map {
     /// Otherwise, updates `m` with `k -> v`.
     ///
     def insertWithKey[k, v](f: (k, v, v) -> v, k: k, v: v, m: Map[k, v]): Map[k, v] =
```

Comment:
> Can I go ahead and note this in the documentation for these functions?

No, because later we plan to add some kind of ordering, and the order of a Set/Map will be determined by that order, and not by the insertion order.

> If an existing key's value is updated in a Map or an element is added to a Set that already exists in the Set, should the corresponding mapping/element be moved to the front of the underlying list of the Map/Set

Yes, because otherwise a Set/Map could grow without bound, correct? For example, if you were to insert 1000 mappings into a Map with the same key then you don't want to store the old mappings.